cpp/oneapi/dal/detail/error_messages.cpp

```diff
@@ -162,6 +162,7 @@ MSG(neighbor_count_lt_one, "Neighbor count lower than one")
 MSG(unknown_distance_type,
     "Custom distances for k-NN is not supported, use one of the predefined distances instead.")
 MSG(distance_is_not_supported_for_gpu, "Only Euclidean distances for k-NN is supported for GPU")
+MSG(unsupported_knn_model, "Provided model is not recognized as oneDAL kNN model")
```

Comment:
I think we need to rephrase it, user may provide search model to the classification inference and error message will be confusing as the user passes oneDAL model. It's better to say something about model incompatibility

BTW: kNN -> k-NN

Reply:
Message has been changed